### ISSUE TITLE
Typed Link field to Hyper Migrate no data migrates

### ISSUE THREAD

**Issue by amici-infotech** [verbb/hyper#221]
### Describe the bug

I have upgraded my site from Craft 4 to Craft 5. I use Neo plugin and Typed Link field. (all latest)

Neo: 5.2.21
Hyper: 2.2.1
Typed Link Field: 3.0.0-beta

### Steps to reproduce

1. Change Typed Link field to Hyper (`./craft hyper/migrate/typed-link-field`
2. Run Migration Command (`./craft hyper/migrate/typed-link-content`)

Migration runs successfully, but content not migrates. When I have checked the code, i think i found some obvious reason why.

https://github.com/verbb/hyper/blob/craft-5/src/migrations/MigrateTypedLinkContent.php#L66
`$settings` contains migrated field content but it is not used to set in content. I changed code to

```
// Get the JSON content for the element with our new content merged in, ready to go
$newContent = $this->getElementContentForField($element, $field);
```
to
```
// Get the JSON content for the element with our new content merged in, ready to go
$newContent = $this->getElementContentForField($element, $field);

$keys = array_keys($newContent);
$newContent[$keys[0]] = Json::encode($settings);
```

and it worked. its just a patch that worked for me so thought to raise it to you to get proper fix.

### Craft CMS version

5.5.7

### Plugin version

2.2.1

### Multi-site?

Yes

### Additional context

All my links are under Neo field.

---
Edit:
Even after my edit above, it not works in `Entry > Neo > Matrix > LINK FIELD` and `Entry > Neo > Neo Child > Matrix > LINK FIELD`

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/hyper:"dev-craft-5 as 2.2.1"`.

**Comment by amici-infotech** [comment]
@engram-design Thanks for the update. ̛It now works on most the cases except 1 place. Steps below

Use Neo > Matrix > Typed Link Field.

Field name was `button2` that i had change to `button` form layout.

![image](https://github.com/user-attachments/assets/20ea7cc0-5a7f-4ed3-b82e-cdffc78bf346)

now this code in `PluginContentMigration.php` file search for original field handle
```
if ($fieldLayout = $element->getFieldLayout()) {
            foreach ($fieldLayout->getCustomFields() as $fieldLayoutField) {
                if ($field->handle === $fieldLayoutField->handle) {
                    $fieldContent[$fieldLayoutField->layoutElement->uid] = $fieldValue;
                }
            }
        }
```

that it never founds as it returns `button` (updated name from layout) and we needed `button2` (original name) as per `MigrateTypedLinkContent.php` line Number 54
```
$field = Craft::$app->getFields()->getFieldById($fieldData['id']);
```

as both button handles dont match, that entry gets skipped and content donot migrates

**Comment by engram-design** [comment]
Good call. Fixed for the next release. To get this early, run `composer require verbb/hyper:"dev-craft-5 as 2.2.1"`.

**Comment by engram-design** [comment]
Fixed in [2.2.2](https://github.com/verbb/hyper/releases/tag/2.2.2)
